test/metabase/lib/filter_test.cljc

```diff
@@ -518,35 +519,105 @@
       {:clause [:= (created-at-with :minute) "2023-10-02T00:00:00.000Z"]
        :name "Created At is Oct 2, 2023, 12:00 AM"}])))
 
-(deftest ^:parallel exclude-date-frontend-filter-display-names-test
+(deftest ^:parallel exclude+equal-date-frontend-filter-display-names-test
   (let [created-at (meta/field-metadata :products :created-at)
         created-at-with #(lib/with-temporal-bucket created-at %1)]
     (check-display-names
-     [{:clause [:!= (created-at-with :day-of-week) "2023-10-02"],
-       :name "Created At excludes Mondays"}
-      {:clause [:!= (created-at-with :day-of-week) "2023-10-02" "2023-10-03" "2023-10-04"],
-       :name "Created At excludes 3 day of week selections"}
-      {:clause [:!= (created-at-with :month-of-year) "2023-01-01"],
-       :name "Created At excludes each Jan"}
-      {:clause [:!= (created-at-with :month-of-year) "2023-01-01" "2023-02-01" "2023-03-01"],
-       :name "Created At excludes 3 month of year selections"}
-      {:clause [:!= (created-at-with :quarter-of-year) "2023-01-03"],
-       :name "Created At excludes Q1 each year"}
-      {:clause [:!= (created-at-with :quarter-of-year) "2023-01-03" "2023-04-03" "2023-07-03"],
-       :name "Created At excludes 3 quarter of year selections"}
-      {:clause [:!= (created-at-with :hour-of-day) 0],
+     [{:clause [:= (created-at-with :hour-of-day) 0]
+       :name "Created At: Hour of day is on 12 AM"}
+      {:clause [:= (lib/get-hour created-at) 0]
+       :name "Created At is at 12 AM"}
```

Comment:
The equality filters have not been tested and they seem strange to me (I suppose because they are not used ATM). I've made an attempt to generate more readable display names for the new filters.